### PR TITLE
[Danger Rules] Documentation for #11280

### PR DESCRIPTION
**Pull request by AliSoftware**
Doc for https://github.com/babylonhealth/babylon-ios/pull/11280

### CODE REVIEW

**Review by danspndl — APPROVED** [review]
✨

File: Cookbook/Technical-Documents/DangerRules.md [hunk 1]
```diff
@@ -120,6 +120,20 @@ This rule will warn you if any part of the pbxproj references an UUID that got a
...
+> * Type: ⚠️ warning
+
+This rule analyses any `*ViewModel.swift` file created or modified in the PR, and try to determine if any `Feedback` static method was declared but not called.
+This is typically to catch cases when you declare a new `Feedback` in your code but forgot to then add it to the state machine.
+
+This rule uses `sourcekitten` to dump the structure of the Swift code. It uses it to finds all the `static func` returning a `Feedback` declared in your code, then searches for all the method calls in your code to see if that list contains calls to all the `Feedback` methods declared. If it finds a `static func … -> Feedback<…>` method that is never called, it emits a warning.
+
+Note that this rule can have false positives in rare cases. Especially if you use constructs like `obj.map(Self.whenBlah)`, it doesn't detect `whenBlah` as a method call because it is encapsulated inside a `map`. This is why this rule is only a warning.
+(If still detects `Feedbacks` called conditionally though –like `if (cond) { feedbacks += Self.whenBlah() }`– and should still catch the most common cases of forgiving to add a newly-created `Feedback` to the state machine.
```
Comment (AliSoftware) [inline Cookbook/Technical-Documents/DangerRules.md:135]:
```suggestion
(If still detects `Feedbacks` called conditionally though –like `if (cond) { feedbacks += Self.whenBlah() }`– and should still catch the most common cases of forgetting to add a newly-created `Feedback` to the state machine.
```

**Review by NSMyself — DISMISSED** [review]

File: Cookbook/Technical-Documents/DangerRules.md [hunk 2]
```diff
@@ -120,6 +120,20 @@ This rule will warn you if any part of the pbxproj references an UUID that got a
...
+> * Type: ⚠️ warning
+
+This rule analyses any `*ViewModel.swift` file created or modified in the PR, and try to determine if any `Feedback` static method was declared but not called.
+This is typically to catch cases when you declare a new `Feedback` in your code but forgot to then add it to the state machine.
+
+This rule uses `sourcekitten` to dump the structure of the Swift code. It uses it to finds all the `static func` returning a `Feedback` declared in your code, then searches for all the method calls in your code to see if that list contains calls to all the `Feedback` methods declared. If it finds a `static func … -> Feedback<…>` method that is never called, it emits a warning.
+
+Note that this rule can have false positives in rare cases. Especially if you use constructs like `obj.map(Self.whenBlah)`, it doesn't detect `whenBlah` as a method call because it is encapsulated inside a `map`. This is why this rule is only a warning.
+(If still detects `Feedbacks` called conditionally though –like `if (cond) { feedbacks += Self.whenBlah() }`– and should still catch the most common cases of forgetting to add a newly-created `Feedback` to the state machine.
```
Comment (NSMyself) [inline Cookbook/Technical-Documents/DangerRules.md:135]:
```suggestion

(It still detects `Feedbacks` called conditionally though – like `if (cond) { feedbacks += Self.whenBlah() }` – and should still catch the most common cases of forgetting to add a newly-created `Feedback` to the state machine)
```

**Review by NSMyself — CHANGES_REQUESTED** [review]
Very minor thing but I think you mean `It` instead of `If`

File: Cookbook/Technical-Documents/DangerRules.md (hunk 2 above)
Comment (AliSoftware) [reply Cookbook/Technical-Documents/DangerRules.md]: Good eye!

**Review by konrad-em — APPROVED** [review]
👍

**Review by NSMyself — APPROVED** [review]

**Pull request merged**